provider.go

```diff
@@ -344,20 +344,11 @@ func (wp *WasmcloudProvider) putLink(l InterfaceLinkDefinition) error {
 
 	wp.lock.Lock()
 	defer wp.lock.Unlock()
-	if l.SourceID == wp.Id {
-		err := wp.putSourceLinkFunc(l)
-		if err != nil {
-			return err
-		}
 
-		wp.sourceLinks[l.Target] = l
+	if l.SourceID == wp.Id {
+		return wp.putSourceLinkFunc(l)
 	} else if l.Target == wp.Id {
-		err := wp.putTargetLinkFunc(l)
-		if err != nil {
-			return err
-		}
-
-		wp.targetLinks[l.SourceID] = l
```

Comment:
Could you comment on why you removed this specific logic? With our example provider we do keep track of links explicitly, however I think in the general case providers shouldn't _have_ to explicitly keep track of links and should be able to store whatever data necessary. Does that make sense? 

Storing the links here is also necessary for the checks in the provider SDK to prevent duplicate links

Reply:
I think I may not be fully understanding the business logic here, or the request to refactor the link logic outside of the provider. This logic, if I'm understanding it correctly, is further down the chain and being stored as part of these functions:

```
func (p *Provider) handleNewSourceLink(link provider.InterfaceLinkDefinition) error {
	log.Println("Handling new source link", link)
	err := p.establishSourceLink(link)
	if err != nil {
		log.Println("Failed to establish source link", link, err)
		p.failedSourceLinks[link.Target] = link
		return err
	}
	p.sourceLinks[link.Target] = link
	return nil
}

func (p *Provider) handleNewTargetLink(link provider.InterfaceLinkDefinition) error {
	log.Println("Handling new target link", link)
	err := p.establishTargetLink(link)
	if err != nil {
		log.Println("Failed to establish target link", link, err)
		p.failedTargetLinks[link.SourceID] = link
		return err
	}
	p.targetLinks[link.SourceID] = link
	return nil
}
```

Reply:
Ah to clarify, I like that the provider SDK itself is keeping track of the links in order to put/delete them from a map and that the developer writing a provider handler doesn't need to keep that map established (even though we do in our example)